domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputIsExactlyEqualToRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,298 @@
+    val inputs = mapOf("f" to createFractionOnly(FRAC_POS_2))
+
+    val matches =
+      isExactlyEqualClassifierProvider.matches(
+        answer = createMixedNumber(MIXED_POS_1),
+        inputs = inputs
+      )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testAnswer_nonNegativeInput_inputWithIncorrectType_throwsException() {
+    val inputs = mapOf("f" to NON_NEGATIVE_VALUE_0)
+
+    val exception = assertThrows(IllegalStateException::class) {
+      isExactlyEqualClassifierProvider.matches(
+        answer = createFractionOnly(FRAC_POS_1),
+        inputs = inputs
+      )
+    }
+
+    assertThat(exception)
+      .hasMessageThat()
+      .contains(
+        "Expected input value to be of type FRACTION not NON_NEGATIVE_INT"
+      )
+  }
+
+  @Test
+  fun testAnswer_testRatio_missingInputF_throwsException() {
+    val inputs = mapOf("y" to createFractionOnly(FRAC_POS_1))
+
+    val exception = assertThrows(IllegalStateException::class) {
+      isExactlyEqualClassifierProvider.matches(
+        answer = createFractionOnly(FRAC_POS_1),
+        inputs = inputs
+      )
+    }
+
+    assertThat(exception)
+      .hasMessageThat()
+      .contains("Expected classifier inputs to contain parameter with name 'f' but had: [y]")
+  }
+
+  private fun createFractionOnly(value: List<Int>): InteractionObject {
+    // Fraction-only numbers imply no whole number.
+    return InteractionObject.newBuilder().setFraction(
+      Fraction.newBuilder()
+        .setIsNegative(value[0] == 1)
+        .setNumerator(value[1])
+        .setDenominator(value[2])
+        .build()
+    ).build()
+  }
+
+  private fun createWholeNumberOnly(value: List<Int>): InteractionObject {
+    // Whole number fractions imply '0/1' fractional parts.
+    return InteractionObject.newBuilder().setFraction(
+      Fraction.newBuilder()
+        .setIsNegative(value[0] == 1)
+        .setWholeNumber(value[1])
+        .setNumerator(0)
+        .setDenominator(1)
+        .build()
+    ).build()
+  }
+
+  private fun createMixedNumber(value: List<Int>): InteractionObject {
+    return InteractionObject.newBuilder().setFraction(
+      Fraction.newBuilder()
+        .setIsNegative(value[0] == 1)
+        .setWholeNumber(value[1])
+        .setNumerator(value[2])
+        .setDenominator(value[3])
+        .build()
+    ).build()
+  }
+
+  private fun createNonNegativeInt(value: Int): InteractionObject {
+    return InteractionObject.newBuilder().setNonNegativeInt(value).build()
+  }
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  private fun setUpTestApplicationComponent() {
+    DaggerFractionInputIsExactlyEqualToRuleClassifierProviderTest_TestApplicationComponent
+      .builder()
+      .setApplication(ApplicationProvider.getApplicationContext()).build().inject(this)
+  }
+
+  // TODO(#89): Move to a common test library.
+  private fun <T : Throwable> assertThrows(type: KClass<T>, operation: () -> Unit): T {
```

Comment:
@rt4914 solving this might be a nice SLoP project idea if we split it up & update the TODOs (or just require that students search for them).

/cc @MohamedMedhat1998 @aggarwalpulkit596 @Sarthak2601

Reply:
This one too [`TODO(#59): Figure out a way to reuse modules instead of needing to re-declare them.` ](https://github.com/oppia/oppia-android/blob/5fcb64712c0952b07cf6ffb2ae38499b636e45c3/app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt#L212) 
But I think it will be tricky.

Reply:
@MohamedMedhat1998 I don't think that's actually possible without something like Dagger Hilt which, as far as I'm aware, doesn't yet have full Gradle support. I was planning to adopt that once Bazel work is completed.